File: Java-Advanced-Jan2020/src/j_DefiningClasses_Exercises/PokemonTrainer/Trainer.java
package j_DefiningClasses_Exercises.PokemonTrainer;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Iterator;
import java.util.List;

public class Trainer {
    private String trainerName;
    private int badges;
    private List<Pokemon> pokemons = new ArrayList<>();

    public Trainer(String trainerName, String pokemonName, String pokemonElement, int pokemonHealth) {
        this.trainerName = trainerName;
        this.badges = 0;
        this.pokemons.add(new Pokemon(pokemonName, pokemonElement, pokemonHealth));
    }

    public int getBadges() {
        return badges;
    }

    public void addPokemonToList(String pokemonName, String pokemonElement, int pokemonHealth) {
        this.pokemons.add(new Pokemon(pokemonName, pokemonElement, pokemonHealth));
    }

    public void pokemonElementCheck(String pokemonElement) {
        for (Pokemon pokemon : pokemons) {
            if (pokemon.getPokemonElement().equals(pokemonElement)) {
                this.badges++;
                return;
            }
        }

        for (int i = 0; i < pokemons.size(); i++) {
            Pokemon current = pokemons.get(i);
            current.setPokemonHealth(current.getPokemonHealth() - 10);
            if (current.getPokemonHealth() <= 0) {
                pokemons.remove(current);
                i--;
            }
        }
    }

    @Override
    public String toString() {
        return this.trainerName + " " + this.badges + " " + this.pokemons.size();
    }
}

File: Programming-for-QA-sep2023/17.ObjectAndClasses/10.PokemonTrainer/Trainer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _10.PokemonTrainer
{
    public class Trainer
    {
        public string Name { get; set; }
        public int NumberOfBadges { get; set; }
        public List<Pokemon> Pokemons { get; set; }

        public Trainer(string name)
        {
            this.Name = name;
            this.NumberOfBadges = 0;
            this.Pokemons = new();
        }

        public override string ToString()
        {
            return $"{this.Name} {this.NumberOfBadges} {this.Pokemons.Count}";
        }


    }
}
